src/containers/ProductPage/ProductPage.js

```diff
@@ -0,0 +1,122 @@
+import Button from "../Button/Button";
+import "./ProductPage.css";
+import { NavLink } from "react-router-dom";
+import { useParams } from "react-router-dom";
+import { useContext } from "react";
+import CartContext from "../../context/CartContext";
+
+
+function ProductPage(products) {
+  const {CartProducts,handleProducts} = useContext(CartContext)
+  let params = useParams();
+  let idItem = params.id;
+  let results = products.Products.data.results;
+  let product = getProduct(results, idItem);
+  if (product === null)
+    return (
+      <div>
+        <h1>Product not Found.</h1>
+      </div>
+    );
+  let title = product.data.name;
+  let img = product.data.mainimage.url;
+  let desc = product.data.short_description;
+  let price = product.data.price;
+  let category = product.data.category.slug;
+  let sku = product.data.sku;
+  let tags = product.tags;
+  let specs = product.data.specs;
+  let id = product.id;
+  let stock = product.data.stock;
+  
+  const getQtyInCart=()=>{
+    let cont=0;
+    CartProducts.forEach((element,i) => {
+      if(id===element.id){
+      cont=element.qty;
+      }
+  });
+  return cont;
+  }
```

Comment:
Maybe we can calculate this in the `CartContext` and send it along with the `CartProducts` and  `HandleProducts` in the provider